SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp

```diff
@@ -170,6 +170,27 @@ void ASpatialTestCrossServerRPC::PrepareTest()
 
 	int NumCubes = CubesLocations.Num();
 
+	// Check actors are ready on each server
+	AddStep(
+		TEXT("Dynamic actor tests: TestReadyActors"), FWorkerDefinition::AllServers,
+		[this, NumCubes]() -> bool {
+			TArray<AActor*> TestCubes;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), ACrossServerRPCCube::StaticClass(), TestCubes);
+
+			bool AreReady = true;
+
+			for (const AActor* Cube : TestCubes)
+			{
+				check(Cube);
+				AreReady &= Cube->IsActorReady();
+			}
+
+			return AreReady;
+		},
+		[this]() {
+			FinishStep();
```

Comment:
I often see a pattern (see below) to wait until a certain condition is true. The main benefit is in adding a separate time limit to the step.
```
AddStep(TEXT("StepName"), FWorkerDefinition, nullptr, nullptr, [this](float)
{
	RequireTrue(/*some condition*/, TEXT("Condition description"));
	FinishStep();
},
StepTimeLimitSeconds);
```